lib/ansible/modules/source_control/github_organization.py

```diff
@@ -0,0 +1,308 @@
+                            method="POST",
+                            headers=headers)
+        if info['status'] == 201:
+            org = _get_organization(module, base_url, auth_header, organization_name)
+            result.update({'changed': True})
+            _merge_organization_result(result, org)
+        else:
+            failure = module.from_json(info['body'])
+            module.fail_json(msg='Failed to create organization: {0}'.format(failure['message']))
+    else:
+        _merge_organization_result(result, existing_org)
+
+    return result
+
+
+def rename_organization(module, auth_header):
+    result = {'changed': False}
+
+    organization_name = module.params['name']
+    new_name = module.params['rename_to']
+
+    if new_name is None:
+        module.fail_json(msg="A name to rename the organization to must be provided")
+
+    base_url = _get_base_url(module)
+    data = {'login': new_name}
+
+    existing_org = _get_organization(module, base_url, auth_header, organization_name)
+    target_org = _get_organization(module, base_url, auth_header, new_name)
+
+    if existing_org is not None:
+        if target_org is None:
+            url = _build_api_url(base_url, 'admin/organizations/{0}'.format(organization_name))
+            headers = {'Authorization': auth_header}
+
+            r, info = fetch_url(module,
+                                url,
+                                data=module.jsonify(data),
+                                method="PATCH",
+                                headers=headers)
+            if info['status'] == 202:
+                # This action is added to a queue, so if successful, report back the change
+                result.update({'changed': True})
+                _merge_organization_result(result, existing_org)
+                result.update({'name': new_name})
+            else:
+                failure = module.from_json(info['body'])
+                module.fail_json(msg='Failed to rename organization: {0}'.format(failure['message']))
+        else:
+            module.fail_json(msg="An organization with the target name already exists, '{0}'".format(new_name))
+
+    return result
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            name=dict(required=True),
+            owner=dict(required=False),
+            rename_to=dict(required=False),
+            base_url=dict(required=True),
+            token=dict(no_log=True, required=True),
+            token_user=dict(required=True),
+            validate_certs=dict(type='bool', default=True),
+            state=dict(choices=['present', 'rename'], default='present')
+        ),
+        supports_check_mode=False,
+    )
+
+    user = module.params['token_user']
+    token = module.params['token']
+    state = module.params['state']
+    auth_header = _get_auth_header(user, token)
+
+    if state == 'present':
+        result = create_organization(module, auth_header)
+    elif state == 'rename':
```

Comment:
I chose to make rename a state due to the slightly odd behaviour of a rename action. See comment above about why I made rename a state.

I am open to suggestions, but I would probably see `state: rename` being used to clean up your GHE organisations as a one time event, rather than something that would be run continuously. A rename action requires the source organization to exist, and the rename to organization to not exist. Once a rename has taken place, the same action cannot be completed, as the source would not exist and target to rename to does.

I thought long and hard on this, but could not see a better way, as this is not really an idempotent action, but an important one to to fix up mistakes or errors.

Reply:
i would require one of id/name for normal operation and require ID if you want to change name, which now can be a 'check if i need to' making it safe to rerun the same task.